### PR TITLE
Fetch channel repodata and patch API responses

### PR DESCRIPTION
**Pull request by jaimergp**
Trying a simple logic first. It hangs locally because my provider won't let me query api.anaconda.org 😬 so... blind attempt!

### CODE REVIEW

**Review by tlambert03 — COMMENTED** [review]
Do you want to add a pull_request trigger that avoids the commit action at the end?  To see if it runs here

**Comment by jaimergp** [comment]
Mm, fetching the repo fails. Is that specific ref needed?

**Comment by tlambert03** [comment]
it's probably that `with: ref: ${{ github.head_ref }}` line... I added that at first to get the `git-auto-commit-action` to work, but it might not be necessary?  try just removing it.  if it fails to trigger a commit on main, we can deal with it then

**Comment by tlambert03** [comment]
https://github.com/stefanzweifel/git-auto-commit-action#checkout-the-correct-branch

**Comment by tlambert03** [comment]
yeah :) was just coming to that conclusion locally.

**Comment by jaimergp** [comment]
Let's try now... I think you can cancel this one, it's probably stuck: https://github.com/napari/npe2api/runs/7676557002?check_suite_focus=true

File: scripts/reindex.py [hunk 1]
```diff
@@ -48,7 +81,13 @@ def conda_data(package_name, channel="conda-forge") -> Tuple[str, dict]:
         url = ANACONDA_ORG.format(channel=channel, package=name)
         with contextlib.suppress(error.HTTPError):
             with request.urlopen(url) as resp:
-                return (package_name, json.load(resp))
+                data = json.load(resp)
+                if repodata:
+                    try:
+                        data = patch_api_data_with_repodata(data, repodata)
```
Comment (tlambert03) [inline scripts/reindex.py:87]: i think you'll either need to return data, or not assign the return value here

**Comment by jaimergp** [comment]
That worked! Let me check a couple things I cannot check locally, then I'll undo and we can review.

**Comment by tlambert03** [comment]
so, one thing I'm not totally understanding yet: when I run this locally, the net result seems to be just that all the `files.dependencies` fields are stripped.  but that's about all I can see.   Will it be on a package by package basis?  do you know of any packages off-hand here that show something other than just stripping `file.dependencies`?

**Comment by jaimergp** [comment]
Nah, I had the logic reversed in L75

**Comment by jaimergp** [comment]
Ok I think that's it @tlambert03

**Comment by tlambert03** [comment]
Awesome!  Will merge so it can do its thing, thanks!

**Comment by tlambert03** [comment]
How's that look: https://github.com/napari/npe2api/commit/eaa016e86d6dc6269a0879762b54f6d7e238a310

**Comment by jaimergp** [comment]
Yep, looks alright! We are not patching any plugins yet, but I can see how in the future we might want patch the version upper bounds on a `napari` spec if they optimistically depended on any version or something like that. Thanks!

**Comment by tlambert03** [comment]
Ok thank you! Smaller files are nice too :)

**Pull request merged**